x-pack/plugins/elastic_assistant/server/lib/langchain/execute_custom_llm_chain/index.ts

```diff
@@ -130,11 +130,16 @@ export const callAgentExecutor: AgentExecutor<true | false> = async ({
         verbose: false,
       })
     : await initializeAgentExecutorWithOptions(tools, llm, {
-        agentType: 'chat-conversational-react-description',
+        agentType: 'structured-chat-zero-shot-react-description',
         memory,
         verbose: false,
+        returnIntermediateSteps: false,
+        handleParsingErrors: 'Try again, paying close attention to the allowed tool input',
```

Comment:
This can also be a function, so I put logs in here while developing. It was not hit once I added the `agentArgs`. However when I was hitting it, it did help to get the agent back on track so I think I should leave it. Here is an example of a run where it was hit: https://smith.langchain.com/public/910b7739-cd9a-401b-9db0-fe8438ff53e5/r

Reply:
FYI - this got merged already with https://github.com/elastic/kibana/pull/182207/files#diff-d9b56949a1c36c72c58e322963a2e97d81485ee83621f804c8849d0eec654a51R133